text/utilities/0000-chmod.md

```diff
@@ -0,0 +1,81 @@
+
+Unfortunately, some applications, like `dist-zip` archives, do need to create directories and files: they run setup scripts during runtime that fail because they can't write folders nor files, since they were run with a runtime user, who can be restricted with some Stacks. 
+
+Similarly, some scripts that need to be run at runtime are only executable by the build time user, and would need to be executable by the runtime user.
+
+## Detailed Explanation
+
+N/A
+
+## Rationale and Alternatives
+
+We have considered solving the issue at different levels:
+
+* At the stack level: reverting to former stacks that define the same user at build time and runtime.
+
+After all, previous stacks like Ubuntu Bionic did not differentiate the runtime user from the build time user, both being named `cnb` with the UID 1000
+
+We began noticing this issue with the latter stacks (some based on Ubuntu Jammy) that guarantee [a more secure environment](https://github.com/buildpacks/rfcs/blob/main/text/0085-run-uid.md) by having the build time user set to UID 1002, and runtime user set to 1000 at runtime.
+
+* At the buildpack level: [implementing special flags in some buildpacks](https://github.com/paketo-buildpacks/dist-zip/pull/174) that would allow users of those buildpacks to make some folders group writable
+
+But then, we discovered that it was not enough, some other flag would need to be implemented to let the build time user run a different set of scripts; and of course, we would need to replicate this behavior in several other buildpacks packaging applications that set themselves up at runtime.
+
+Allowing the user to keep on using the same secure Stacks, and unmodified buildpacks, using a chmod buildpack at the end of their build seems like the best solution
+
+## Implementation
+
+The chmod buildpack would be added last to builders and would be activated if a specific flag is set.
+
+Two flags, or environment variables, are suggested for configuring the `chmod` buildpack.
+
+1. An environment variable named `BP_CHMOD_MAPPING` would be configured with a comma-separated map of folders and their desired owner and permissions.
+
+For example:
+
+`pack build -e BP_CHMOD_MAPPING="/workspace:0750:r,/workspace/bin:0755" myimage`
+
+would activate the chmod buildpack and 
+
+* make it change recursively the `/workspace` folder of the runtime image to have permissions `0750`
+* make it change non-recursively the `/workspace/bin` folder of the runtime image to have permissions `0775` (making all these folder files executable)
+
+2. An environment variable named `BP_CHMOD_MAPPING_FILE` would be configured with the location of a file defining the mapping (convenient for users configuring numerous mappings)
+
+For example, the combination of:
+
+`pack build -e BP_CHMOD_MAPPING_FILE=chmod-mappings.txt myimage`
```

Comment:
Why not auto-detect on this file? It seems unlikely that this file will exist for any other use-case and we're not shadowing existing filenames (like `pom.xml`). Asking the user to type extra env vars seems cumbersome and unnecessary.

Reply:
I have been making an assumption here 🤦 basically that `BP_CHMOD_MAPPING_FILE` would default to a sensible file name and that if users include that file, then this would automatically work like you're saying @robdimsdale. Like how we handle pom.xml/build.gradle search in those buildpacks.

@anthonydahanne Is that what you were thinking? Maybe we should clarify here? 

My concern with not having the env variable is that invariably someone will ask if they can put the file somewhere else and change where the buildpack looks for it.

Reply:
Sure. I think that's fair. I like the compromise of providing sensible defaults with extensibility where necessary.

Reply:
There was some strong feedback in the working group meeting about ensuring it is hard/impossible for users to accidentally execute this buildpack with dangerous permissions (e.g. `chmod -R 755 /workspace`).

So, it sounds like it would be prudent to keep the environment variable pointing to the file as a required argument.

Reply:
I don't personally have strong opinions on this, I'm just representing the feedback from the working group meeting.

Reply:
ok we keep setting the environment variable as a mandatory step; it will make it more foolproof.

Reply:
I'll actually add a note about this